### PR TITLE
doc: incorporate Go reference document changes for Alpha release

### PR DESCRIPTION
**Pull request by jyyi1**

### CODE REVIEW

File: network/delegate_packet_proxy.go [hunk 1]
```diff
@@ -19,14 +19,14 @@ import (
 	"sync/atomic"
 )
 
-// DelegatePacketProxy is a PacketProxy that forwards calls (like NewSession) to another PacketProxy. To create a
-// DelegatePacketProxy with the default PacketProxy, use NewDelegatePacketProxy. To change the underlying PacketProxy,
-// use SetProxy.
+// DelegatePacketProxy is a [PacketProxy] that forwards calls (like NewSession) to another [PacketProxy]. To create a
```
Comment (fortuna) [inline network/delegate_packet_proxy.go:22]: You may want to call out the intent of the Delegate, which is to replace the packetProxy without changing the original reference.

**Review by fortuna — APPROVED** [review]

File: network/delegate_packet_proxy.go (hunk 1 above)
Comment (jyyi1) [reply network/delegate_packet_proxy.go]: added

**Review by fortuna — APPROVED** [review]

File: network/delegate_packet_proxy.go [hunk 2]
```diff
@@ -19,14 +19,14 @@ import (
...
+// DelegatePacketProxy with the default PacketProxy, use [NewDelegatePacketProxy]. To change the underlying
+// [PacketProxy], use SetProxy.
 //
-// Note: After changing the underlying PacketProxy, only new NewSession calls will be routed to the new PacketProxy.
-// Existing sessions will not be affected.
+// Note: After the underlying [PacketProxy] is changed, only new NewSession calls will be routed to the new
```
Comment (maddyhof) [inline network/delegate_packet_proxy.go:27]: should be: [New Session]

File: network/delegate_packet_proxy.go [hunk 3]
```diff
@@ -19,14 +19,14 @@ import (
...
+// [PacketProxy]. Existing sessions will not be affected.
 //
-// Multiple goroutines may invoke methods on a DelegatePacketProxy simultaneously.
+// Multiple goroutines can simultaneously invoke methods on a DelegatePacketProxy.
```
Comment (maddyhof) [inline network/delegate_packet_proxy.go:30]: [DelegatePacketProxy]

File: network/doc.go [hunk 4]
```diff
@@ -14,10 +14,10 @@
 
 /*
 The network package defines interfaces and provides utilities for network layer (OSI layer 3) functionalities. For
-example, the [IPDevice] interface can be used to read and write IP packets from a physical or virtual network device.
+example, you can use the [IPDevice] interface to read and write IP packets from a physical or virtual network device.
 
-In addition, user-space network stack implementations are also included in the sub-packages (such as
-network/lwip2transport) that can translate raw IP packets into TCP/UDP flows. You can implement a [PacketProxy]
-to handle UDP traffic, and a [transport.StreamDialer] to handle TCP traffic.
+In addition, the sub-packages include user-space network stack implementations (such as network/lwip2transport) that
```
Comment (maddyhof) [inline network/doc.go:19]: [network/lwip2transport]

File: network/packet_proxy.go [hunk 5]
```diff
@@ -34,14 +34,14 @@ type PacketProxy interface {
 }
 
 // PacketRequestSender sends UDP request packets to the [PacketProxy]. It should be implemented by the [PacketProxy],
```
Comment (maddyhof) [inline network/packet_proxy.go:36]: [PacketRequestSender]

File: network/packet_proxy.go [hunk 6]
```diff
@@ -34,14 +34,14 @@ type PacketProxy interface {
...
-// who must implement the WriteTo method to process the request packets. PacketRequestSender is typically called by an
-// upstream component (such as a network stack). Once the Close method is called, there will be no more requests sent
-// to the sender, and all resources can be freed.
+// which must implement the WriteTo method to process the request packets. PacketRequestSender is typically called by
```
Comment (maddyhof) [inline network/packet_proxy.go:37]: [WriteTo]

Comment (maddyhof) [reply network/packet_proxy.go:37]: [PacketRequestSender]

File: network/packet_proxy.go [hunk 7]
```diff
@@ -53,13 +53,13 @@ type PacketRequestSender interface {
 	Close() error
 }
 
-// PacketResponseReceiver receives UDP response packets from the [PacketProxy]. It is usually implemented by the
-// upstream component (such as a network stack). When creating a new UDP session, a valid instance
-// PacketResponseReceiver is passed to the PacketProxy.NewSession function. Then the [PacketProxy] must use this
-// instance to send UDP responses all the time; and it must call the Close function to indicate there will be no more
-// responses send to this receiver.
+// PacketResponseReceiver receives UDP response packets from the [PacketProxy]. It is usually implemented by an
```
Comment (maddyhof) [inline network/packet_proxy.go:56]: [PacketResponseReceiver]

File: network/packet_proxy.go [hunk 8]
```diff
@@ -53,13 +53,13 @@ type PacketRequestSender interface {
...
+// upstream component (such as a network stack). When a new UDP session is created, a valid instance of
+// PacketResponseReceiver is passed to the PacketProxy.NewSession function. The [PacketProxy] must then use this
```
Comment (maddyhof) [inline network/packet_proxy.go:58]: [PacketResponseReceiver]

File: network/packet_proxy.go [hunk 9]
```diff
@@ -53,13 +53,13 @@ type PacketRequestSender interface {
...
+// instance to send UDP responses. It must then call the Close function to indicate that there will be no more
```
Comment (maddyhof) [inline network/packet_proxy.go:59]: [Close]

File: network/packet_proxy.go [hunk 10]
```diff
@@ -53,13 +53,13 @@ type PacketRequestSender interface {
...
+// responses sent to this receiver.
 //
-// Multiple goroutines may invoke methods on a PacketResponseReceiver simultaneously.
+// Multiple goroutines can simultaneously invoke methods on a PacketResponseReceiver.
```
Comment (maddyhof) [inline network/packet_proxy.go:62]: [PacketResponseReceiver]

File: transport/packet.go [hunk 11]
```diff
@@ -26,11 +26,11 @@ type PacketEndpoint interface {
 	Connect(ctx context.Context) (net.Conn, error)
 }
 
-// UDPEndpoint is a [PacketEndpoint] that connects to the given address via UDP
+// UDPEndpoint is a [PacketEndpoint] that connects to the specified address using UDP
```
Comment (maddyhof) [inline transport/packet.go:29]: [UDPEndpoint]

Comment (maddyhof) [reply transport/packet.go:29]: Add a period at the end of the sentence.

File: transport/packet.go [hunk 12]
```diff
@@ -42,7 +42,7 @@ func (e UDPEndpoint) Connect(ctx context.Context) (net.Conn, error) {
 	return e.Dialer.DialContext(ctx, "udp", e.Address)
 }
 
-// PacketDialerEndpoint is a [PacketEndpoint] that connects to the given address using the given [PacketDialer].
+// PacketDialerEndpoint is a [PacketEndpoint] that connects to the given address using the specified [PacketDialer].
```
Comment (maddyhof) [inline transport/packet.go:45]: [PacketDialerEndpoint]

File: transport/packet.go [hunk 13]
```diff
@@ -91,9 +91,9 @@ type boundPacketConn struct {
 var _ net.Conn = (*boundPacketConn)(nil)
 
 // Dial implements [PacketDialer].Dial.
```
Comment (maddyhof) [inline transport/packet.go:93]: [Dial]

Comment (maddyhof) [reply transport/packet.go:93]: [[PacketDialer.Dial]. (move the bracket to include the .Dial part, per Hilary

File: transport/packet.go [hunk 14]
```diff
@@ -91,9 +91,9 @@ type boundPacketConn struct {
...
-// The address is a host:port and the host must be a full IP address (not [::]) or a domain
-// The address must be supported by the WriteTo call of the PacketConn
-// returned by the PacketListener. For instance, a [net.UDPConn] only supports IP addresses, not domain names.
+// The address is in "host:port" format and the host must be either a full IP address (not "[::]") or a domain.
+// The address must be supported by the WriteTo call of the [net.PacketConn] returned by the [PacketListener].
```
Comment (maddyhof) [inline transport/packet.go:95]: [WriteTo]

File: transport/stream.go [hunk 15]
```diff
@@ -20,14 +20,13 @@ import (
 	"net"
 )
 
-// StreamConn is a net.Conn that allows for closing only the reader or writer end of
-// it, supporting half-open state.
+// StreamConn is a [net.Conn] that allows for closing only the reader or writer end of it, supporting half-open state.
```
Comment (maddyhof) [inline transport/stream.go:23]: [StreamConn]

File: transport/stream.go [hunk 16]
```diff
@@ -59,8 +58,8 @@ func (dc *duplexConnAdaptor) CloseWrite() error {
 	return dc.StreamConn.CloseWrite()
 }
 
-// WrapDuplexConn wraps an existing [StreamConn] with new Reader and Writer, but
-// preserving the original [StreamConn.CloseRead] and [StreamConn.CloseWrite].
+// WrapConn wraps an existing [StreamConn] with a new [io.Reader] and [io.Writer], but preserves the original
```
Comment (maddyhof) [inline transport/stream.go:61]: [WrapConn]

File: transport/stream.go [hunk 17]
```diff
@@ -59,8 +58,8 @@ func (dc *duplexConnAdaptor) CloseWrite() error {
...
+// [StreamConn].CloseRead and [StreamConn].CloseWrite.
```
Comment (maddyhof) [inline transport/stream.go:62]: Hilary had these as [StreamConn.CloseRead] and [StreamConn.CloseWrite].

File: transport/stream.go [hunk 18]
```diff
@@ -70,13 +69,14 @@ func WrapConn(c StreamConn, r io.Reader, w io.Writer) StreamConn {
 	return &duplexConnAdaptor{StreamConn: conn, r: r, w: w}
 }
 
-// StreamEndpoint represents an endpoint that can be used to established stream connections (like TCP) to a fixed destination.
+// StreamEndpoint represents an endpoint that can be used to establish stream connections (like TCP) to a fixed
```
Comment (maddyhof) [inline transport/stream.go:72]: [StreamEndpoint]

File: transport/stream.go [hunk 19]
```diff
@@ -96,7 +96,8 @@ func (e *TCPEndpoint) Connect(ctx context.Context) (StreamConn, error) {
 	return conn.(*net.TCPConn), nil
 }
 
-// StreamDialerEndpoint is a [StreamEndpoint] that connects to the given address using the given [StreamDialer].
+// StreamDialerEndpoint is a [StreamEndpoint] that connects to the specified address using the specified
```
Comment (maddyhof) [inline transport/stream.go:99]: [StreamDialerEndpoint]

File: transport/stream.go [hunk 20]
```diff
@@ -70,13 +69,14 @@ func WrapConn(c StreamConn, r io.Reader, w io.Writer) StreamConn {
...
-// StreamEndpoint represents an endpoint that can be used to established stream connections (like TCP) to a fixed destination.
+// StreamEndpoint represents an endpoint that can be used to establish stream connections (like TCP) to a fixed
+// destination.
 type StreamEndpoint interface {
 	// Connect establishes a connection with the endpoint, returning the connection.
 	Connect(ctx context.Context) (StreamConn, error)
 }
 
-// TCPEndpoint is a [StreamEndpoint] that connects to the given address using the given [StreamDialer].
+// TCPEndpoint is a [StreamEndpoint] that connects to the specified address using the specified [StreamDialer].
```
Comment (maddyhof) [inline transport/stream.go:79]: [TCPEndpoint]

**Review by maddyhof — COMMENTED** [review]
Reviewed against the doc from Hilary and identified a bunch of formatting things that were not implemented. Nothing is critical but I flagged as many as I could. Happy to review again once those changes are in and confirm that nothing else is amiss.

File: network/delegate_packet_proxy.go (hunk 2 above)
Comment (jyyi1) [reply network/delegate_packet_proxy.go:27]: interface method names should not be wrapped in `[]`

File: network/doc.go (hunk 4 above)
Comment (jyyi1) [inline network/doc.go:19]:
```suggestion
In addition, the sub-packages include user-space network stack implementations (such as [network/lwip2transport]) that
```

File: network/packet_proxy.go (hunk 9 above)
Comment (jyyi1) [reply network/packet_proxy.go:59]: interface method names should not be wrapped in `[]`

File: transport/packet.go (hunk 11 above)
Comment (jyyi1) [inline transport/packet.go:29]:
```suggestion
// UDPEndpoint is a [PacketEndpoint] that connects to the specified address using UDP.
```

File: transport/packet.go (hunk 13 above)
Comment (jyyi1) [reply transport/packet.go:93]: interface method names should not be wrapped in `[]`

File: transport/stream.go (hunk 17 above)
Comment (jyyi1) [reply transport/stream.go:62]: interface method names should not be wrapped in `[]`

File: network/packet_proxy.go [hunk 21]
```diff
@@ -34,14 +34,14 @@ type PacketProxy interface {
...
 // PacketRequestSender sends UDP request packets to the [PacketProxy]. It should be implemented by the [PacketProxy],
-// who must implement the WriteTo method to process the request packets. PacketRequestSender is typically called by an
-// upstream component (such as a network stack). Once the Close method is called, there will be no more requests sent
-// to the sender, and all resources can be freed.
+// which must implement the WriteTo method to process the request packets. PacketRequestSender is typically called by
+// an upstream component (such as a network stack). After the Close method is called, there will be no more requests
+// sent to the sender, and all resources can be freed.
 //
-// Multiple goroutines may invoke methods on a PacketRequestSender simultaneously.
+// Multiple goroutines can simultaneously invoke methods on a PacketRequestSender.
```
[the remark this answers is not in the document]
Comment (jyyi1) [reply network/packet_proxy.go:41]: Type names should not be wrapped in `[]` in the documentation that describes the type itself.

File: network/packet_proxy.go (hunk 10 above)
Comment (jyyi1) [reply network/packet_proxy.go:62]: Type names should not be wrapped in `[]` in the documentation that describes the type itself.

File: transport/packet.go (hunk 11 above)
Comment (jyyi1) [reply transport/packet.go:29]: Type names should not be wrapped in `[]` in the documentation that describes the type itself. Added "." at the end.

File: transport/stream.go (hunk 15 above)
Comment (jyyi1) [reply transport/stream.go:23]: Type names should not be wrapped in `[]` in the documentation that describes the type itself.

File: transport/stream.go (hunk 18 above)
Comment (jyyi1) [reply transport/stream.go:72]: Type names should not be wrapped in `[]` in the documentation that describes the type itself.

File: transport/stream.go (hunk 19 above)
Comment (jyyi1) [reply transport/stream.go:99]: Type names should not be wrapped in `[]` in the documentation that describes the type itself.

**Review by jyyi1 — COMMENTED** [review]
Thanks @maddyhof , these are all [Godoc specific formats](https://tip.golang.org/doc/comment#doclinks), and there are some restrictions: for example, only type names can be wrapped by `[]` but not method names. Also typically we don't wrap a type with `[]` in the documentation that describes the type itself (example, the `Closer` is not wrapped in `[]` in [`io.Closer`](https://pkg.go.dev/io#Closer)). Similarly, we won't add backticks "`" to most of the names, unless it is a local function parameter.

File: network/doc.go (hunk 4 above)
Comment (jyyi1) [reply network/doc.go]: updated

File: network/packet_proxy.go (hunk 5 above)
Comment (jyyi1) [reply network/packet_proxy.go:36]: Type names should not be wrapped in [] in the documentation that describes the type itself.

File: transport/address.go [hunk 22]
```diff
@@ -35,12 +35,12 @@ func (a *domainAddr) String() string {
 var _ net.Addr = (*domainAddr)(nil)
 
 // MakeNetAddr returns a [net.Addr] based on the network and address.
```
[the remark this answers is not in the document]
Comment (jyyi1) [reply transport/address.go:37]: Type names should not be wrapped in [] in the documentation that describes the type itself.

**Comment by maddyhof** [comment]
> 

As discussed, feel free to ignore my comments that don't make sense in the Go style, and reject Hilary's suggestions in the doc so that we know which things aren't going to be implemented.

**Pull request merged**